module/TaskManagement/src/TaskManagement/Controller/MailController.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace TaskManagement\Controller;
+
+use ZFX\Rest\Controller\HATEOASRestfulController;
+use TaskManagement\Service\NotifyMailListener;
+use TaskManagement\Service\TaskService;
+use TaskManagement\Entity\Task;
+use TaskManagement;
+use TaskManagement\Entity\TaskMember;
+use Zend\View\Model\ViewModel;
+
+
+class MailController extends HATEOASRestfulController {
```

Comment:
Capisco il criterio che hai utilizzato, ma la scelta se "schedulare" un invio o renderlo "on demand" con un bottone è una responsabilità di Front End / User Experience. A livello di API / Back End deve essere disponibile l'invio della notifica. E le notifiche, a livello di Back End, non sono diverse. Per questo sarebbe utile centralizzarne la gestione in unico punto.
La decisione su *chi* possa inviare una specifica notifica è, come tutti gli aspetti di controllo degli accessi, invece una responsabilità di Back End centralizzata nella definizione di una ACL nel nostro caso. E qui, come ti dicevo, è utile identificare ogni notifica come risorsa a sé stante.

Spero di averti chiarito meglio perché centralizzerei nel `RemindersController` la gestione della notifica che hai realizzato.